test/testsuite/lantest.c

```diff
@@ -127,162 +172,561 @@ static unsigned long timediff(void)
+        }
+    } else {
+        if (csv_output) {
+            fprintf(stdout, "Sum of all AFP OPs = %d,%.0f,,\n",
+                    TOTAL_AFP_OPS,
+                    column_sums[MEASURE_TIME_MS]);
+        } else {
+            char summary_str[100];
+            snprintf(summary_str, sizeof(summary_str), "Sum of all AFP OPs = %d",
+                     TOTAL_AFP_OPS);
+            fprintf(stdout, "%-66s %7.0f %6s %6s\n",
+                    summary_str,
+                    column_sums[MEASURE_TIME_MS], "", "");
         }
+    }
 
-        /* Calculate standard deviation if we have multiple valid iterations */
-        std_dev = 0.0;
+#else
+
+    if (csv_output) {
+        fprintf(stdout, "Sum of all AFP OPs = %d,%.0f,,\n",
+                TOTAL_AFP_OPS,
+                column_sums[MEASURE_TIME_MS]);
+    } else {
+        char summary_str[100];
+        snprintf(summary_str, sizeof(summary_str), "Sum of all AFP OPs = %d",
+                 TOTAL_AFP_OPS);
+        fprintf(stdout, "%-66s %7.0f %6s %6s\n",
+                summary_str,
+                column_sums[MEASURE_TIME_MS], "", "");
+    }
 
-        if (valid_count > 1) {
-            double sum_sq_diff = 0.0;
+#endif
 
-            for (i = 0; i < Iterations_save; i++) {
-                if ((*results)[i][test] > 0) {
-                    double diff = (double)((*results)[i][test]) - (double)avg;
-                    sum_sq_diff += diff * diff;
-                }
-            }
+    /* Print aggregates summary */
+    if (!csv_output) {
+        fprintf(stdout, "\nAggregates Summary:\n");
+        fprintf(stdout, "-------------------\n");
+        /* Calculate average time per AFP OP */
+        double avg_time_per_op = column_sums[MEASURE_TIME_MS] / (double)TOTAL_AFP_OPS;
+        fprintf(stdout, "Average Time per AFP OP: %.3f ms\n", avg_time_per_op);
+#ifdef __linux__
 
-            std_dev = sqrt(sum_sq_diff / (valid_count - 1));
+        if (io_monitoring_enabled) {
+            /* Calculate ratio of AFPD Reads to all AFP OPs */
+            double read_ratio = column_sums[MEASURE_AFPD_READ_IO] / (double)TOTAL_AFP_OPS;
+            fprintf(stdout, "Average AFPD Reads per AFP OP: %.3f\n", read_ratio);
+            /* Calculate ratio of AFPD Writes to all AFP OPs */
+            double write_ratio = column_sums[MEASURE_AFPD_WRITE_IO] / (double)TOTAL_AFP_OPS;
+            fprintf(stdout, "Average AFPD Writes per AFP OP: %.3f\n", write_ratio);
         }
 
-        /* Display test result */
-        fprintf(stderr, "Test %d: %s", test + 1, resultstrings[test]);
+#endif
+        fprintf(stdout, "\n");
+    }
+}
+
+static void displayresults(void)
+{
+    int i, test, maxindex, minindex;
+    unsigned long long sum;
+    unsigned long max, min;
+    int measure_type;
 
-        if (Iterations_save > 1) {
-            fprintf(stderr, "\n\t  Average: %6llu ms", avg);
+    /* Eliminate runaways for all measurement types */
+    if (Iterations_save > 5) {
```

Comment:
Random thought: Do we want to document in the man page this logic for eliminating outliers? And perhaps indicate in the test results that outliers have been eliminated from the results?